BuildingBlocks/src/BuildingBlocks.Application/FluentValidation/ValueInValidator.cs

```diff
@@ -59,11 +62,28 @@ protected override string GetDefaultMessageTemplate(string errorCode)
 
 public static class ValidatorExtensions
 {
+    private static readonly ConcurrentDictionary<Type, MethodInfo> IS_VALID_CACHE = new();
+
     public static IRuleBuilderOptions<T, TProperty> In<T, TProperty>(this IRuleBuilder<T, TProperty> ruleBuilder,
         params TProperty[] validOptions)
     {
         return ruleBuilder
             .SetValidator(new ValueInValidator<T, TProperty>(validOptions))
             .WithErrorCode(GenericApplicationErrors.Validation.InvalidPropertyValue().Code);
     }
+
+    public static IRuleBuilderOptions<T, string?> ValidId<T, TId>(this IRuleBuilder<T, string?> ruleBuilder) where TId : StronglyTypedId
+    {
+        var method = IS_VALID_CACHE.FirstOrDefault(x => x.Key == typeof(TId)).Value;
```

Comment:
Can't you access a ConcurrentDictionary with an indexer?

```suggestion
        var method = IS_VALID_CACHE[x.Key];
```

Reply:
I think you meant:
`var method = IS_VALID_CACHE[typeof(TId)];`
I could, but this throws when there are no matches, so we would be replacing the `FirstOrDefault` with a subsequent `null` check for a `try / catch` block where we use the indexer.
Thoughts?

Reply:
True. Then use `TryGetValue` instead:

```c#
if(IS_VALID_CACHE.TryGetValue(x.Key, out var method)
```